<launch>
  <arg name="teleop" default="true"/>

  <!-- params -->
  <param name="/pedsim/scene_file" value="$(find pedsim_simulator)scenarios/corridor_fke.xml" type="string"/>
  <param name="/use_sim_time" value="false"/>

  <!-- main simulator node -->
  <node name="pedsim_simulator" pkg="pedsim_simulator" type="simulator" output="screen"/>

  <!-- Rviz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pedsim_simulator)/rviz/social_contexts_activities.rviz"/>


  <group ns="/pedbot/control">
    <!-- robot driving controller -->
    <node name="driving_controller" type="simulate_diff_drive_robot" pkg="pedsim_simulator" output="screen">
        <param name="pose_initial_x" value="4.0"/>
        <param name="pose_initial_y" value="4.0"/>
        <param name="pose_initial_theta" value="0.0"/>  <!-- default settings x 4.0, y 4.0 theta 0.7 -->
    </node>

    <!-- RQT robot steering
    <node pkg="rqt_robot_steering" name="rqt_robot_steering" type="rqt_robot_steering" if="$(arg teleop)">
        <remap from="/cmd_vel" to="/pedbot/control/cmd_vel"/>
    </node> -->
  </group>

  <param name="/pedsim_point_clouds/local_width" value="8.0" type="double"/>
  <param name="/pedsim_point_clouds/local_height" value="8.0" type="double"/>

  <!-- NODES -->
  <node name="pedsim_point_clouds" pkg="pedsim_point_clouds" type="pedsim_point_clouds" output="screen"/>

</launch>
